shared/src/packages/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -1,21 +1,45 @@
 import joi from 'joi';
 
-import { UserValidationMessage } from '../enums/enums.js';
 import { type UserSignUpRequestDto } from '../types/types.js';
+import { Regexp as REGEXP } from './enums/regexp.enum.js';
+import { UserValidationMessage } from './enums/user-validation-message.enum.js';
 
-const userSignUp = joi.object<UserSignUpRequestDto, true>({
+const passwordPattern = new RegExp(REGEXP.PASSWORD);
+const namePattern = new RegExp(REGEXP.NAME);
+const phonePattern = new RegExp(REGEXP.PHONE);
+
+const userSignUpRules = {
+  phone: joi.string().trim().required().pattern(phonePattern).messages({
+    'string.empty': UserValidationMessage.REQUIRED,
+    'string.pattern.base': UserValidationMessage.PHONE_NOT_VALID,
+  }),
+  password: joi.string().trim().required().pattern(passwordPattern).messages({
+    'string.empty': UserValidationMessage.REQUIRED,
+    'string.pattern.base': UserValidationMessage.PASSWORD_NOT_VALID,
+  }),
   email: joi
     .string()
     .trim()
+    .min(5)
+    .max(254)
```

Comment:
Let's combine all user validation constants in one enum

Reply:
moved to user-validation-rules.enum